src/services/codefixes/importFixes.ts

```diff
@@ -228,6 +230,27 @@ namespace ts.codefix {
         return { description, changes, commands };
     }
 
+    function getSymbolExportInfoForSymbol(symbol: Symbol, moduleSymbol: Symbol, importingFile: SourceFile, program: Program, host: LanguageServiceHost): SymbolExportInfo {
+        const compilerOptions = program.getCompilerOptions();
+        const mainProgramInfo = getInfoWithChecker(program.getTypeChecker());
+        if (mainProgramInfo) {
+            return mainProgramInfo;
+        }
+        const autoImportProvider = host.getPackageJsonAutoImportProvider?.()?.getTypeChecker();
+        return Debug.checkDefined(autoImportProvider && getInfoWithChecker(autoImportProvider), `Could not find symbol in specified module for code actions`);

```

Comment:
so this asserts if the host can't produce an auto import provider can't be produced or a type checker?

Reply:
Not exactly; it asserts that the exported symbol was in fact found to be exported by its module symbol, according to either the main program checker or the AutoImportProvider checker. This is basically redoing work during the completions _details_ request that was done during the previous completions _list_ request, so it should always be found, but of course there’s no guarantee of that in the types, hence the assertion. (It’s also the sort of assumption that could plausibly be violated by a bizarre sequence of edits triggering some race condition, or just wrong code, so we do see assertions like this come up in telemetry every once in a while.)